File: backend/bank_sync/aa_providers.py
"""
Setu Account Aggregator (AA) provider configuration.

This file is the single source of truth for supported AA entity handles.
Update this list when new AAs are onboarded to Setu's production environment.

Verified sandbox AA handles (from GET /v2/fips aaWiseSuccessRate on 2026-07-31):
  - onemoney
  - anumati

Production may include additional providers — add them here, not inline in code.
"""

AA_PROVIDERS = [
    {"handle": "onemoney",  "label": "OneMoney"},
    {"handle": "anumati",   "label": "Anumati"},
]

# Derived set for fast validation
VALID_AA_HANDLES = {p["handle"] for p in AA_PROVIDERS}

File: frontend/src/utils/aaProviders.js
/**
 * AA Provider configuration for the frontend dropdown.
 *
 * These are populated from the backend's GET /api/v1/bank-sync/aa-providers/ endpoint.
 * This file contains the static fallback list in case the API call fails.
 * Update aa_providers.py on the backend — do NOT maintain this list separately.
 *
 * Confirmed valid Setu sandbox handles (2026-07-31):
 *   onemoney, anumati
 */
export const AA_PROVIDERS = [
  { handle: 'onemoney', label: 'OneMoney' },
  { handle: 'anumati',  label: 'Anumati'  },
];
